qiskit/transpiler/passes/routing/sabre_swap.py

```diff
@@ -210,11 +210,17 @@ def run(self, dag):
 
         dag_list = []
         for node in dag.topological_op_nodes():
+            cargs = []
+            if node.op.condition is not None and not node.cargs:
+                cargs = [self._clbit_indices[x] for x in dag._bits_in_condition(node.op.condition)]
+            else:
+                cargs = [self._clbit_indices[x] for x in node.cargs]
```

Comment:
Does this work with a conditional measurement (e.g. an operation that has both a `condition` and non-empty `cargs`)?

Reply:
Heh, I didn't test that because I assumed they were xor. Let me throw together a quick test and I think you're right I'll need to tweak the condition here somewhat

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/8727/commits/99d756b3710e3652a4c0553ce0442ed91caa4293